provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/contract/ContractApiHttp.java

```diff
@@ -183,6 +184,7 @@ private HttpRequest createRequest(URL url, HttpMethod method, Map<String, String
         request.setHeaderField(HEADER_FIELD_NAME_ACCEPT, HEADER_FIELD_VALUE_ACCEPT);
         request.setHeaderField(HEADER_FIELD_NAME_CONTENT_TYPE, HEADER_FIELD_VALUE_CONTENT_TYPE);
         request.setHeaderField(HEADER_FIELD_NAME_CHARSET, HEADER_FIELD_VALUE_CHARSET);
+        request.setHeaderField(HEADER_FIELD_NAME_CONTENT_LENGTH, payload != null ? String.valueOf(payload.length) : "0");
```

Comment:
If this is not a hard requirement, why was it throwing failure only occasionally?

Reply:
I'm not sure. Service folks didn't answer me when I asked if it was a hard requirement. This seems like a safe change to make regardless, though